### DISCUSSION TITLE
Why isn't there a M2A relationship between the x_editor_nodes model and the node_records?

### DISCUSSION THREAD

**Discussion by jdjoshuadavison** [formfcw/directus-extension-flexible-editor#20] [Q&A] [answered]
These are my models:

- pages
- resources
- appeals
- page_editor_nodes
- related_resource
- related_appeal

The `pages` model contains a flexible editor that I've set up relation nodes (`related_resource`, `related_appeal`) to allow users to reference `resources` and `appeals` within the editor.

But while there is a relationship between `pages` and `page_editor_nodes`, and a relationship between `related_resources` and `resources`, there is no relationship between `related_resources` and `page_editor_nodes`.

Currently:
![flexible_editor_relationships](https://github.com/formfcw/directus-extension-flexible-editor/assets/34007858/38e3b4ae-e6ac-4eb0-bc51-513c535c6f51)

Why not?:
![flexible_editor_relationships_m2a](https://github.com/formfcw/directus-extension-flexible-editor/assets/34007858/b655a273-38ae-4dce-aefe-114bbe9fa34e)

It's not a big deal, but if the relationship existed it would be possible to add fields to the other records so that there would be a "path" in the GUI from a resource, or an appeal, all the way to the page it is referenced on (resource -> related_resouce -> page_editor_node -> page). I don't know whether we would do this or not, but it would be possible.

**Comment by formfcw** [comment, marked as answer]
Thank you for your question, but I'm afraid only Directus can answer that for you. This has nothing to do with Flexible Editor as it is just an interface. I also recognized this and it is related to the fact that Directus does not offer an A2M field – an inverse relationship to M2A.

**Reply by jdjoshuadavison** [reply]
Thanks for the answer - you're completely right. My diagram is wrong. What you require is an any-to-many, not a many-to-any, and that does not exist. Thank you - I knew there had to be a reason! :)
